gcore/gdalhashsetbandblockcache.cpp

```diff
@@ -47,23 +46,32 @@
 
 class GDALHashSetBandBlockCache final : public GDALAbstractBandBlockCache
 {
-    struct BlockComparator
+    struct BlockHash
     {
-        // Do not change this comparator, because this order is assumed by
-        // tests like tiff_write_133 for flushing from top to bottom, left
-        // to right.
-        bool operator()(const GDALRasterBlock *const &lhs,
-                        const GDALRasterBlock *const &rhs) const
+        std::size_t operator()(GDALRasterBlock *const &b) const noexcept
```

Comment:
The left shift could potentially trigger a sanitizer warning (what's the rationale for it by the way? a code comment could be useful)
```suggestion
        CPL_NOSANITIZE_UNSIGNED_INT_OVERFLOW
        std::size_t operator()(GDALRasterBlock *const &b) const noexcept
```

Reply:
Hash functions are magic and I have never spent much time on them. I stole this from elsewhere expecting that the original author spent more time than me. If we're pretty sure that we're working with 64 bits as size_t (which is probably most platforms now), you could probably just take the two numbers and force them into a uint64_t and run std::hash<uint64_t> on that. But if someone else wants to spend time finding a great hash function that would be fine. I just like simple.